relayer-cli/src/components.rs

```diff
@@ -45,47 +25,38 @@ impl JsonTracing {
         let builder = FmtSubscriber::builder()
             .with_target(false)
             .with_env_filter(filter)
-            .with_writer(std::io::stderr as StdWriter)
+            .with_writer(std::io::stderr)
             .with_ansi(use_color)
             .with_thread_ids(true)
-            .json()
-            .with_filter_reloading();
-
-        let filter_handle = builder.reload_handle();
+            .json();
 
         let subscriber = builder.finish();
         subscriber.init();
 
-        Ok(Self { filter_handle })
+        Ok(Self)
     }
 }
 
 #[derive(Component, Debug)]
-pub struct PrettyTracing {
-    filter_handle: Handle<EnvFilter, PrettyFormatter>,
-}
+pub struct PrettyTracing;
 
 impl PrettyTracing {
     /// Creates a new [`PrettyTracing`] component
-    #[allow(trivial_casts)]
     pub fn new(cfg: GlobalConfig) -> Result<Self, FrameworkError> {
         let filter = build_tracing_filter(cfg.log_level.to_string())?;
 
         // Construct a tracing subscriber with the supplied filter and enable reloading.
         let builder = FmtSubscriber::builder()
             .with_target(false)
             .with_env_filter(filter)
-            .with_writer(std::io::stderr as StdWriter)
+            .with_writer(std::io::stderr)
```

Comment:
Just a note: This code should be tested at a basic level in the E2E python scripts, which rely on the JSON output. It would still be useful to run some CLI commands with and without `--json` to make sure we're not changing the structure of the output here.

We can do that, however, in the release PR (which will close #1648): on that occasion, we also would like to update the guide, which requires re-running the CLI anyway.

Reply:
This is just a stylistic change, the cast did nothing.

Reply:
It was required on a previous version of rustc, glad to see it's not needed anymore :)

Reply:
Ah hm, this means we've implicitly bumped our MSRV.
If we care about such things, it may make sense to add a `clippy.toml` and maintain the MSRV there.

Reply:
Yeah let's do that!